borsh/src/de/mod.rs

```diff
@@ -516,6 +511,70 @@ pub mod hashes {
             Ok(result)
         }
     }
+
+    #[cfg(feature = "de_strict_order")]
+    pub mod strict_order {
```

Comment:
It creates a risk of the implementations diverging. Let's not duplicate the deserialization logic and just introduce the feature flag inside the existing deserialization implementations as you did for BTreeSet.